openstack/02a-MAClow/IEEE802154E.c

```diff
@@ -550,11 +551,24 @@ port_INLINE void activity_synchronize_endOfFrame(PORT_RADIOTIMER_WIDTH capturedT
       ieee154e_vars.dataReceived->l2_frameType = ieee802514_header.frameType;
       ieee154e_vars.dataReceived->l2_dsn       = ieee802514_header.dsn;
       memcpy(&(ieee154e_vars.dataReceived->l2_nextORpreviousHop),&(ieee802514_header.src),sizeof(open_addr_t));
-      
-      // toss the IEEE802.15.4 header -- this does not include IEs as they are processed 
+
+      // This is the case where we first hear an EB and do not yet know the ASN used

```

Comment:
Even if K1 is secret, you are relying on leap of faith during join as you are not synced to the network. You are taking ASN from the EB you want to authenticate and then feeding that back into the authentication routing for the EB to pass. We can read the ASN this way *only* during join, all other EBs should be authenticated based on the implicit ASN, not the one present in them. Do you agree?